robottelo/ui/base.py

```diff
@@ -64,31 +65,71 @@ def find_element(self, locator):
             )
         return None
 
-    def search_entity(self, element_name, element_locator, search_key=None,
-                      katello=None, button_timeout=15, result_timeout=15):
-        """Uses the search box to locate an element from a list of elements."""
-        search_key = search_key or 'name'
+    def _search_locator(self):
+        """Specify element name locator which should be use in search procedure
+
+        """
+        raise NotImplementedError(
+            'Subclasses must return locator of element to search')
+
+    def _search_init(self, search_key=None, is_katello=None, button_timeout=15,
+                     result_timeout=15):
+        """Initialize every optional parameter for search procedure which can
+        be overridden for any unique case"""
+        if self.is_entity_delete:
```

Comment:
Could add a comment explaining the logic here? Why do we override the value for `button_timeout`? I'm sure there's a reason, it's just not clear to me from reading the code :)